nixos/modules/tasks/network-interfaces.nix

```diff
@@ -417,7 +417,11 @@ in
         network node hostname (uname --nodename) the option
         boot.kernel.sysctl."kernel.hostname" can be used as a workaround (but
         the 64 character limit still applies).
+
+        WARNING: Do not use underscores(_).
```

Comment:
Not sure if this really helps because according to the description underscores are not allowed anyway (it's an undocumented "feature"). But who knows maybe it'll help someone.

But something like: `WARNING: Do not use underscores (_) or you might run into unexpected issues (depending on the software that you use).` might make it more clear.

Reply:
https://github.com/NixOS/nixpkgs/pull/138978#issuecomment-947239026

Reply:
Yes, not sure how that relates to my comment though.

I'm referring to the description/documentation (not the implementation):
> This means that the hostname must start with a letter or digit,
        end with a letter or digit, and have as interior characters only
        letters, digits, and hyphen.

Reply:
> But something like: `WARNING: Do not use underscores (_) or you might run into unexpected issues (depending on the software that you use).` might make it more clear.

done